agent/config/file_watcher.go

```diff
@@ -0,0 +1,222 @@
+package config
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"path/filepath"
+	"time"
+
+	"github.com/fsnotify/fsnotify"
+	"github.com/hashicorp/go-hclog"
+)
+
+const timeoutDuration = 200 * time.Millisecond
+
+type FileWatcher struct {
+	watcher          *fsnotify.Watcher
+	configFiles      map[string]*watchedFile
+	EventsCh         chan *WatcherEvent
+	logger           hclog.Logger
+	reconcileTimeout time.Duration
+	cancel           context.CancelFunc
+	done             chan interface{}
+}
+
+type watchedFile struct {
+	modTime time.Time
+}
+
+type WatcherEvent struct {
```

Comment:
Maybe better to name this `FileWatcherEvent` as it is related to a `FileWatcher` and is in the `agent/config` package instead of its own?